smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/util/AbstractConfigGenerator.kt

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.kotlin.codegen.rendering.util
+
+import software.amazon.smithy.codegen.core.SymbolReference
+import software.amazon.smithy.kotlin.codegen.core.CodegenContext
+import software.amazon.smithy.kotlin.codegen.core.KotlinWriter
+import software.amazon.smithy.kotlin.codegen.core.withBlock
+import software.amazon.smithy.kotlin.codegen.model.PropertyTypeMutability
+import software.amazon.smithy.kotlin.codegen.model.propertyTypeMutability
+
+/**
+ * Re-usable base class for generating some type that only contains configuration.
+ * e.g. roughly something shaped like below
+ *
+ * ```
```

Comment:
**Nit**: ` ```kotlin `

Reply:
I don't think this matters for KDoc FWIW